python/lsst/ip/isr/crosstalk.py

```diff
@@ -23,17 +23,441 @@
+        """
+        X_FLIP = {lsst.afw.cameraGeom.ReadoutCorner.LL: False,
+                  lsst.afw.cameraGeom.ReadoutCorner.LR: True,
+                  lsst.afw.cameraGeom.ReadoutCorner.UL: False,
+                  lsst.afw.cameraGeom.ReadoutCorner.UR: True}
+        Y_FLIP = {lsst.afw.cameraGeom.ReadoutCorner.LL: False,
+                  lsst.afw.cameraGeom.ReadoutCorner.LR: False,
+                  lsst.afw.cameraGeom.ReadoutCorner.UL: True,
+                  lsst.afw.cameraGeom.ReadoutCorner.UR: True}
+
+        output = image[amp.getBBox() if isTrimmed else amp.getRawDataBBox()]
+        thisAmpCorner = amp.getReadoutCorner()
+        targetAmpCorner = ampTarget.getReadoutCorner()
+
+        # Flipping is necessary only if the desired configuration doesn't match what we currently have
+        xFlip = X_FLIP[targetAmpCorner] ^ X_FLIP[thisAmpCorner]
+        yFlip = Y_FLIP[targetAmpCorner] ^ Y_FLIP[thisAmpCorner]
+        self.log.debug("Extract amp: %s %s %s %s",
+                       amp.getName(), ampTarget.getName(), thisAmpCorner, targetAmpCorner)
+        return lsst.afw.math.flipImage(output, xFlip, yFlip)
+
+    def calculateBackground(self, mi, badPixels=["BAD"]):
+        """Calculate median background in image
+
+        Getting a great background model isn't important for crosstalk correction,
+        since the crosstalk is at a low level. The median should be sufficient.
+
+        Parameters
+        ----------
+        mi : `lsst.afw.image.MaskedImage`
+            MaskedImage for which to measure background.
+        badPixels : `list` of `str`
+            Mask planes to ignore.
+        Returns
+        -------
+        bg : `float`
+            Median background level.
+        """
+        mask = mi.getMask()
+        stats = lsst.afw.math.StatisticsControl()
+        stats.setAndMask(mask.getPlaneBitMask(badPixels))
+        return lsst.afw.math.makeStatistics(mi, lsst.afw.math.MEDIAN, stats).getValue()
+
+    def subtractCrosstalk(self, thisExposure, sourceExposure=None, crosstalkCoeffs=None,
+                          badPixels=["BAD"], minPixelToMask=45000,
+                          crosstalkStr="CROSSTALK", isTrimmed=False,
+                          backgroundMethod="None"):
+        """Subtract the crosstalk from thisExposure, optionally using a different source.
+
+        We set the mask plane indicated by ``crosstalkStr`` in a target amplifier
+        for pixels in a source amplifier that exceed `minPixelToMask`. Note that
+        the correction is applied to all pixels in the amplifier, but only those
+        that have a substantial crosstalk are masked with ``crosstalkStr``.
+
+        The uncorrected image is used as a template for correction. This is good
+        enough if the crosstalk is small (e.g., coefficients < ~ 1e-3), but if it's
+        larger you may want to iterate.
+
+        This method needs unittests (DM-18876), but such testing requires
```

Comment:
Again, add a TODO please

Reply:
Removed as those tickets have been resolved, and updated unittests were added previously.